applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c

```diff
@@ -1232,9 +1229,12 @@ static int discover_sink(struct bt_conn *conn)
 	int ret = 0;
 	static uint8_t discover_index;
 
-	audio_sink_discover_param[discover_index].func = discover_sink_cb;
-	audio_sink_discover_param[discover_index].dir = BT_AUDIO_DIR_SINK;
-	ret = bt_bap_unicast_client_discover(conn, &audio_sink_discover_param[discover_index]);
+	num_caps = 0;
```

Comment:
What if two devices are being discovered at the same time? Say you start discovery on device A, it has 5 caps. While you are discovering these the num_caps is increasing by one each time. Say you get to cap#3 and then device B is connected and we want to do discovery of its capabilities. This will then reset the counter back to 0 while still waiting for cap#4 and cap#5

Reply:
I think num_eps and num_caps should be within the headset struct at the top, and not global variables.

Reply:
Good point and yes they should.